src/algorithms/contractions/ctmrg_contractions.jl

```diff
@@ -578,59 +878,283 @@ function full_infinite_environment(
         E_8[χ11 D21 D22; χ_in]
 end
 
-# Renormalization contractions
-# ----------------------------
+"""
+    full_infinite_environment(
+        quadrant1::T, quadrant2::T, quadrant3::T, quadrant4::T
+    ) where {T<:AbstractTensorMap{<:ElementarySpace,2,2}}
+    function full_infinite_environment(
+        half1::T, half2::T
+    ) where {T<:AbstractTensorMap{<:ElementarySpace,2,2}}
+    full_infinite_environment(C_1, C_2, C_3, C_4, E_1, E_2, E_3, E_4, E_5, E_6, E_7, E_8,
+                              partfunc_1::P, partfunc_2::P) where {P<:PartitionFunctionTensor}
+    full_infinite_environment(C_1, C_2, E_1, E_2, E_3, E_4, x,
+                              partfunc_1::P, partfunc_2::P, partfunc_3::P, partfunc_4::P) where {P<:PartitionFunctionTensor}
+    full_infinite_environment(x, C_1, C_2, E_1, E_2, E_3, E_4,
+                              partfunc_1::P, partfunc_2::P, partfunc_3::P, partfunc_4::P) where {P<:PartitionFunctionTensor}
 
-# corners
+Contract four quadrants (enlarged corners) to form a full-infinite environment.
 
-"""
-    renormalize_corner(quadrant::AbstractTensorMap{S,3,3}, P_left, P_right)
+```
+    |~~~~~~~~~| -- |~~~~~~~~~|
+    |quadrant1|    |quadrant2|
+    |~~~~~~~~~| -- |~~~~~~~~~|
+      |    |         |     |
+                     |     |
+      |    |         |     |
+    |~~~~~~~~~| -- |~~~~~~~~~|
+    |quadrant4|    |quadrant3|
+    |~~~~~~~~~| -- |~~~~~~~~~|
+```
 
-Apply projectors to each side of a quadrant.
+In the same manner two halfs can be used to contract the full-infinite environment.
 
 ```
-    |~~~~~~~~| -- |~~~~~~|
-    |quadrant|    |P_left| --
-    |~~~~~~~~| == |~~~~~~|
-     |    ||
-    [P_right]
-        |
+    |~~~~~~~~~~~~~~~~~~~~~~~~|
+    |         half1          |
+    |~~~~~~~~~~~~~~~~~~~~~~~~|
+      |    |         |     |
+                     |     |
+      |    |         |     |
+    |~~~~~~~~~~~~~~~~~~~~~~~~|
+    |         half2          |
+    |~~~~~~~~~~~~~~~~~~~~~~~~|
 ```
-"""
-function renormalize_corner(quadrant::AbstractTensorMap{S,3,3}, P_left, P_right) where {S}
-    return @autoopt @tensor corner[χ_in; χ_out] :=
-        P_right[χ_in; χ1 D1 D2] * quadrant[χ1 D1 D2; χ2 D3 D4] * P_left[χ2 D3 D4; χ_out]
-end
 
-"""
-    renormalize_northwest_corner((row, col), enlarged_envs::CTMRGEnv, P_left, P_right)
-    renormalize_northwest_corner(quadrant::AbstractTensorMap{S,3,3}, P_left, P_right) where {S}
-    renormalize_northwest_corner(E_west, C_northwest, E_north, P_left, P_right, ket::PEPSTensor, bra::PEPSTensor=ket)
+The environment can also be contracted directly from all its constituent tensors.
 
-Apply `renormalize_corner` to the enlarged northwest corner.
-Alternatively, provide the constituent tensors and perform the complete contraction.
+```
+    C_1 --  E_2       --  E_3       -- C_2
```

Comment:
I'd be careful with the numbering in the pictures, since this might be interpreted as indexing, which is not correct now. I think it is anyways a lot more readable if we use compass directions as subscripts

Reply:
While I agree, since this diagram spans multiple row-col sites properly annotating it with both direction and site indices might become a bit unwieldy. To be honest I wouldn't want to update it since I don't really know what's going on.

I merged all the docstrings so we only need to change it in one place, but since this the diagram was already like this before this PR maybe we can address this in a separate PR?